### DISCUSSION TITLE
DB per user or viceversa? Best practices to handle large set of data

### DISCUSSION THREAD

**Discussion by sergiupopescu199** [apache/couchdb#3511] [Q&A]
Hi, I created a 5 node couchdb cluster and it's working very well! I really love it! Thank you very much for you hard work! 🙏

So I'm not experienced with couchdb but I have this questions. I use it to store metedata of blockchain transactions. I have a database where I store user credentials and other where I store all the transaction metadata for all the users.

What is the best way of taking advantage of couchdb capabilities? 
- make a separate database to store transaction metadata per user ? (could have even thousands or millions of users)  
- use one partioned database to store all the transactions metadata?
- or there is another way to handle this?
- how many databases couchdb can handle? 
- how many documents a database can handle?

Any help will be incredibly useful for me, thanks in advance

**Comment by florianb** [comment]
Hi @sergiupopescu199 - i haven't faced that use case yet. All i can remember having seen and discussed at Slack was, that a database per user approach is perfectly fine.

The two important documentation pages are:
https://docs.couchdb.org/en/stable/intro/security.html#authentication-database
https://docs.couchdb.org/en/stable/config/couch-peruser.html

As far as i know (and i am a beginning user) there are not many limits (one single example might be https://docs.couchdb.org/en/stable/config/couchdb.html#couchdb/max_document_size) as you might know from other databases. 

When some things grow too big things might degrade in performance. F.e. if the single documents grow out too much. But i don't know of any important additional things to know when getting started which aren't written down in the docs and i don't know of any "hidden" show stoppers which would put a scale up on risk.

I hope this helps a bit.
